Arrangement_on_surface_2/include/CGAL/draw_arrangement_2.h

```diff
@@ -0,0 +1,522 @@
+// Copyright (c) 2012
+// Utrecht University (The Netherlands),
+// ETH Zurich (Switzerland),
+// INRIA Sophia-Antipolis (France),
+// Max-Planck-Institute Saarbruecken (Germany),
+// and Tel-Aviv University (Israel).  All rights reserved.
+//
+// This file is part of CGAL (www.cgal.org)
+//
+// $URL$
+// $Id$
+// SPDX-License-Identifier: LGPL-3.0-or-later OR LicenseRef-Commercial
+//
+//
+// Author(s): Efi Fogel <efifogel@gmail.com>
+
+#ifndef CGAL_DRAW_ARRANGEMENT_2_H
+#define CGAL_DRAW_ARRANGEMENT_2_H
+
+#include <unordered_map>
+#include <cstdlib>
+#include <random>
+
+#include <CGAL/Qt/Basic_viewer_qt.h>
+
+#ifdef CGAL_USE_BASIC_VIEWER
+
+#include <type_traits>
+
+
+#if defined(BOOST_MSVC)
+// workaround for using boost::hana with MSVC without warnings
+#pragma push_macro("__cplusplus")
+#undef __cplusplus
+#define __cplusplus _MSVC_LANG
+#include <boost/hana.hpp>
```

Comment:
I added a workaround so that we don't get warnings while using `boost::hana`. It is a new boost dependencies which explain why we get new warnings. I think we even have some errors. It would be best IMO if we can avoid using it by relying on `MPL` and `BOOST_MPL_HAS_XXX` (if I correctly get your hana usage).